modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache

```diff
@@ -3,7 +3,18 @@
  */{{#description}}
 @ApiModel(description = "{{{description}}}"){{/description}}
 {{>generatedAnnotation}}{{#discriminator}}{{>typeInfoAnnotation}}{{/discriminator}}{{>xmlAnnotation}}
-public class {{classname}} {{#parent}}extends {{{parent}}}{{/parent}} {{#serializableModel}}implements Serializable{{/serializableModel}} {
+{{^parent}}
+{{#hateoas}}
+{{#jackson}}
+@JsonPropertyOrder({
```

Comment:
This is to ensure that the links part of the JSON object is the last field in the object (it's a minor issue but the standard in the HATEOAS examples appears to be for the links to come last). E.g:

`        {
            "field1": "value1",
            "field2": "value2",
            "field3": null,
            "field4": null,
            "_links": {
                "self": {
                    "href": "http://localhost:8080/api/v1/thingy-api/value1"
                }
            }
        }`

Without this ordering the _links part appeared as the first item in the JSON object.

Reply:
I'm pretty sure that HATEOAS doesn't have such convention. If it were the case, spring-hateoas would deal with that. Can you remove this part ?

Reply:
I think I need to take another look at this. All the examples seem to show the links as being the last thing but in my case without adding explicit ordering they end up being the first thing. I do agree that this ordering shouldn't be needed though.

Reply:
Just an update on where I've got so far:
I feel like I must be doing something wrong because the Jackson library seems to always add the parent fields before the current fields when serializing and object that extends another class. So since the generated class extends the ResourceSupport class the links field on the ResourceSupport class will always come before the generated class.

However all of the Spring HATEOAS examples show the links as the last field in the JSON that's produced so I feel like I must be doing something differently.

Reply:
Don't you need to add `@EnableHypermediaSupport` annotation ?

Reply:
That doesn't seem to have made any difference. I'm going to try and work out what's different between my application and the example in https://spring.io/guides/gs/rest-hateoas/ which does have the links at the end.

Reply:
Urgh... what's going on with the ordering is a bit difficult to explain:

All of the ordering logic exists with in the com.fasterxml.jackson.databind.introspect.POJOPropertiesCollector class.

The example in the link above the serialized object is using the @JsonCreator annotation on its constructor, and for whatever reason the jackson library makes sure that any fields from the @JsonCreator constructor come before all the other fields. Since the links are coming from the super class rather than the constructor they come last in this case.

However... if you change the class so it doesn't use @JsonCreator and you ensure it doesn't use any @JsonProperty annotation then the links will be placed at the end because the jackson library ensure that any fields that are renamed using the @JsonProperty come last. In this case since the ResourceSupport class has an @JsonProperty annotation on the getLinks method the links will be renamed and so will come last.

The generated code uses getters with the @JsonProperty annotation on all them regardless of whether they're being renamed as opposed to using the @JsonCreator annotation. This means that all the fields including the links are being renamed and so they maintain the same order as before they were being renamed where the parent fields come first.

It's probably just coincidence that all of the examples have the links at the end and is probably because they examples are all relatively simple so don't tend to make use of the @JsonProperty annotation.

**The tldr; is that the ordering is based on the inner workings on the Jackson library. I think it doesn't really matter where the links appear but I think it probably looks a little bit neater to have them at the end.

I'm happy to remove this ordering if you insist, and happy to discuss further if you'd like.**